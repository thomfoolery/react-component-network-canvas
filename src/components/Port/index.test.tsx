import React from "react";
import renderer from "react-test-renderer";

import {Port} from "./index";

describe("Port", () => {
  it("renders correctly", () => {
    const node = {
      id: "node-1",
      position: {
        x: 0,
        y: 0,
      },
      inputPorts: [{id: "node-1-input-1"}],
      outputPorts: [{id: "node-1-output-1"}],
    };
    const port = {id: "node-1-input-1"};
    const type = "input";
    const onMouseUp = () => null;
    const onMouseDown = () => null;

    const tree = renderer
      .create(
        <Port
          node={node}
          port={port}
          type={type}
          onMouseUp={onMouseUp}
          onMouseDown={onMouseDown}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
