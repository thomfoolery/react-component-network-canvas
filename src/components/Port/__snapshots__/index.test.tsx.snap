// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Port renders correctly 1`] = `
<div
  id="Port-node-1-input-1"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
/>
`;
